go-controller/pkg/config/config_test.go

```diff
@@ -1172,6 +1182,48 @@ enable-pprof=true
 		gomega.Expect(Gateway.MasqueradeIPs.V6OVNMasqueradeIP.String()).To(gomega.Equal("fd68::1"))
 
 	})
+	It("returns an error when the v4 transit switch subnet specified is invalid", func() {
+		app.Action = func(ctx *cli.Context) error {
+			_, err := InitConfig(ctx, kexec.New(), nil)
+			gomega.Expect(err).To(gomega.MatchError("invalid transit switch v4 subnet specified, subnet: foobar: error: invalid CIDR address: foobar"))
+			return nil
+		}
+		cliArgs := []string{
+			app.Name,
+			"-cluster-manager-v4-transit-switch-subnet=foobar",
+		}
+		err := app.Run(cliArgs)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	})
+	It("returns an error when the v6 transit switch subnet specified is invalid", func() {
+		app.Action = func(ctx *cli.Context) error {
+			_, err := InitConfig(ctx, kexec.New(), nil)
+			gomega.Expect(err).To(gomega.MatchError("invalid transit switch v6 subnet specified, subnet: 100.89.0.0/16: error: <nil>"))
+			return nil
+		}
+		cliArgs := []string{
+			app.Name,
+			"-cluster-manager-v6-transit-switch-subnet=100.89.0.0/16",
```

Comment:
do we need a test to specify an invalid v4 CIDR?

Reply:
I did this because similar test exists for join and masquerade subnets. We can remove this if it does not add any value.

Reply:
ah no no I mean do we need to also add a new test case for "invalid CIDRs", this test is fine, should we do another where the CIDR is actually invalid?

Reply:
I might be interpreting this requirement wrong but I think below test is already checking whether a random string with CIDR  flag exhibits an error or not.

https://github.com/ovn-org/ovn-kubernetes/pull/4074/files#diff-e7fc36054fe67be368da5d34e133743af74cefbaed5ab51b8952bc3bfa8432ceR1185

Reply:
Ack